### PR TITLE
feat: remove secretegenerator controller

### PR DESCRIPTION
**Pull request by zdtsw**
- we do not need that controller since OAuthClient will be created by Gateway service
- move secret releated function into pkg

## Merge criteria

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [ ] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [ ] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work
- [ ] The developer has run the integration test pipeline and verified that it passed successfully

### E2E test suite update requirement

When bringing new changes to the operator code, such changes are by default required to be accompanied by extending and/or updating the E2E test suite accordingly.

To opt-out of this requirement:
1. **Please inspect the [opt-out guidelines](https://github.com/opendatahub-io/opendatahub-operator/blob/main/docs/e2e-update-requirement-guidelines.md)**, to determine if the nature of the PR changes allows for skipping this requirement
2. If opt-out is applicable, provide justification in the dedicated `E2E update requirement opt-out justification` section below
3. Check the checkbox below:
- [x] Skip requirement to update E2E test suite for this PR
4. Submit/save these changes to the PR description. This will automatically trigger the check.

#### E2E update requirement opt-out justification

remove old function so no add new e2e for it

## Summary by CodeRabbit

* **Refactor**
  * Secret-generation logic moved to a new cluster package; runtime secret generation for auth components preserved.
* **Documentation**
  * Design notes updated; legacy secret-generator README removed.
* **Chores**
  * Legacy secret-generator controller removed and internal wiring updated to use the new generator.
  * Removed an import that triggered startup side-effects from the main entrypoint.
* **Tests**
  * Legacy secret-generator tests removed; new tests added for the cluster-based generator.

### CODE REVIEW

File: pkg/cluster/secret_test.go [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			secret, err := cluster.NewSecret(tc.secretName, tc.secretType, tc.complexity)
+
+			// Check for errors
+			if err != nil {
+				t.Fatalf("NewSecret returned unexpected error: %v", err)
+			}
```
Comment (davidebianchi) [inline pkg/cluster/secret_test.go]: Shouldn't we use require, or assert? Also for the other assertions

```suggestion
			 require.NoError(t, err)
```

File: pkg/cluster/secret_test.go [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+	}
+
+	// Check that all secrets are unique
+	for i := range numSecrets {
+		for j := i + 1; j < numSecrets; j++ {
+			if secrets[i] == secrets[j] {
+				t.Errorf("Secrets %d and %d are identical: %q", i, j, secrets[i])
+			}
+		}
+	}
```
Comment (davidebianchi) [inline pkg/cluster/secret_test.go]: If we really want this test, it could be done using a map[string]bool instead of a slice, and verifying that the map length is equal to the numSecrets

File: pkg/cluster/secret_test.go [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+}
+
+// TestNewSecretEdgeCases tests edge cases for secret generation.
+func TestNewSecretEdgeCases(t *testing.T) {
```
Comment (davidebianchi) [inline pkg/cluster/secret_test.go]: Those tests could be added to the `TestNewSecretWithRandomType` fn, without another test case

File: pkg/cluster/secret_test.go [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+	}
+}
+
+// TestSecretDefaultComplexity tests the default complexity constant.
+func TestSecretDefaultComplexity(t *testing.T) {
+	expectedComplexity := 16
+
+	if cluster.SecretDefaultComplexity != expectedComplexity {
+		t.Errorf("Expected cluster.SecretDefaultComplexity to be %d, got %d", expectedComplexity, cluster.SecretDefaultComplexity)
+	}
```
Comment (davidebianchi) [inline pkg/cluster/secret_test.go]: cluster.SecretDefaultComplexity is a const, why this assertion?

File: pkg/cluster/secret_test.go [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+
+	// Test using the constant
+	secret, err := cluster.NewSecret("default-secret", "random", cluster.SecretDefaultComplexity)
+	if err != nil {
+		t.Fatalf("Failed to generate secret with default complexity: %v", err)
+	}
+
+	if len(secret.Value) != cluster.SecretDefaultComplexity {
+		t.Errorf("Expected secret length %d, got %d", cluster.SecretDefaultComplexity, len(secret.Value))
+	}
```
Comment (davidebianchi) [inline pkg/cluster/secret_test.go]: Could be added at tests done in `TestNewSecretWithRandomType`

File: pkg/cluster/secret_test.go [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+}
+
+// BenchmarkNewSecretRandom benchmarks random secret generation.
+func BenchmarkNewSecretRandom(b *testing.B) {
```
Comment (davidebianchi) [inline pkg/cluster/secret_test.go]: If we want to perform benchmarks, we should also save the output somewhere and check if across different changes they get worse. What do you think?

File: pkg/cluster/secret_test.go (hunk 2 above)
Comment (zdtsw) [reply pkg/cluster/secret_test.go]: probably not needed, i will remove this test

File: pkg/cluster/secret_test.go (hunk 6 above)
Comment (zdtsw) [reply pkg/cluster/secret_test.go]: i will remove these two tests, was original added when gateway was doing benchmark, but it  should not be part of the unit test for secret here

File: pkg/cluster/secret_test.go (hunk 4 above)
Comment (zdtsw) [reply pkg/cluster/secret_test.go]: removed this change and moved test into `TestNewSecretWithRandomType`

**Comment by zdtsw** [comment]
did change test removed most of tests which are not needed: now only on oauth, random and invalid. should be enough to cover two basic functions

File: pkg/cluster/secret_test.go [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+)
+
+// TestNewSecretWithRandomType tests the creation of a random-type secret.
+func TestNewSecretWithRandomType(t *testing.T) {
+	secret, err := cluster.NewSecret("test-secret", "random", cluster.SecretDefaultComplexity)
+	if err != nil {
+		t.Errorf("NewSecret returned unexpected error: %v", err)
+	}
```
Comment (davidebianchi) [inline pkg/cluster/secret_test.go]: Why not use assert/require?

Comment (zdtsw) [reply pkg/cluster/secret_test.go]: rewrite it to gomega, maybe we should have a follow up PR to change `cluster_config_test.go `as well ?

Comment (zdtsw) [reply pkg/cluster/secret_test.go]: https://github.com/opendatahub-io/opendatahub-operator/pull/2769 updated here

**Review by davidebianchi — COMMENTED** [review]
/lgtm
